src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
@baroquebobcat A use case that transitive exports might be useful is:
say we have 4 targets A B C and D. C extends D, B extends C, and A uses B.
In that case, if exports is transitive, then C exports D, B exports C will be enough.
But if exports is not transitive, then B has to exports both C and D. Since B only has C in its dependencies list, exporting D is a little unexpected.